src/services/tests/invalidate_caches.rs

```diff
@@ -0,0 +1,163 @@
+    mock_cache
+        .expect_invalidate_pattern()
+        .times(1)
+        .return_const(())
+        .with(eq(String::from("*0x65df8a1e5a40703d9c67d5df6f9b552d3830faf0507c3d7350ba3764d3a68621*")))
+        .in_sequence(&mut sequence);
+
+    let context = TestContext(mock_cache);
+
+    invalidate_caches(&context, &payload).unwrap();
+}
+
+#[test]
+fn invalidate_executed_multisig_transaction_payload() {
+    let payload = Payload {
+        address: "0x1230B3d59858296A31053C1b8562Ecf89A2f888b".to_string(),
+        details: Some(PayloadDetails::ExecutedMultisigTransaction(
+            ExecutedMultisigTransaction {
+                safe_tx_hash: "0x65df8a1e5a40703d9c67d5df6f9b552d3830faf0507c3d7350ba3764d3a68621".to_string(),
+                tx_hash: "0x0ebb2c317f55c96469e0ed2014f5833dc02a70b42f0ac52f4630938900caa698".to_string(),
+            })),
+    };
+
+    let mut mock_cache = MockCache::new();
+    let mut sequence = Sequence::new();
+    mock_cache
+        .expect_fetch()
+        .times(0);
+    mock_cache
+        .expect_create()
+        .times(0);
+    mock_cache
+        .expect_invalidate()
+        .times(0);
+    mock_cache
+        .expect_invalidate_pattern()
+        .times(1)
+        .return_const(())
+        .with(eq(String::from("*0x1230B3d59858296A31053C1b8562Ecf89A2f888b*")))
+        .in_sequence(&mut sequence);
+    mock_cache
+        .expect_invalidate_pattern()
+        .times(1)
+        .return_const(())
+        .with(eq(String::from("*0x65df8a1e5a40703d9c67d5df6f9b552d3830faf0507c3d7350ba3764d3a68621*")))
+        .in_sequence(&mut sequence);
+
+    let context = TestContext(mock_cache);
+
+    invalidate_caches(&context, &payload).unwrap();
+}
+
+#[test]
+fn invalidate_pending_multisig_transaction_payload() {
+    let payload = Payload {
+        address: "0x1230B3d59858296A31053C1b8562Ecf89A2f888b".to_string(),
+        details: Some(PayloadDetails::PendingMultisigTransaction(
+            PendingMultisigTransaction {
+                safe_tx_hash: "0x65df8a1e5a40703d9c67d5df6f9b552d3830faf0507c3d7350ba3764d3a68621".to_string(),
+            })),
+    };
+
+    let mut mock_cache = MockCache::new();
+    let mut sequence = Sequence::new();
+    mock_cache
+        .expect_fetch()
+        .times(0);
+    mock_cache
+        .expect_create()
+        .times(0);
+    mock_cache
+        .expect_invalidate()
+        .times(0);
+    mock_cache
+        .expect_invalidate_pattern()
+        .times(1)
+        .return_const(())
+        .with(eq(String::from("*0x1230B3d59858296A31053C1b8562Ecf89A2f888b*")))
+        .in_sequence(&mut sequence);
+    mock_cache
+        .expect_invalidate_pattern()
+        .times(1)
+        .return_const(())
+        .with(eq(String::from("*0x65df8a1e5a40703d9c67d5df6f9b552d3830faf0507c3d7350ba3764d3a68621*")))
+        .in_sequence(&mut sequence);
+
+    let context = TestContext(mock_cache);
+
+    invalidate_caches(&context, &payload).unwrap();
+}
```

Comment:
I like the tests already ... for completeness we should also test the other payloads in the future. 

Are there parametrized tests in rust?

Reply:
I will keep this in mind for the next one :D I would have to read into it.